tests/test_keras_integration.py

```diff
@@ -0,0 +1,277 @@
+        new_model = DummyModel.from_pretrained(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+
+        # Check the reloaded model's weights match the original model's weights
+        self.assertTrue(tf.reduce_all(tf.equal(new_model.weights[0], model.weights[0])))
+
+        # Check a new model's weights are not the same as the reloaded model's weights
+        another_model = DummyModel()
+        another_model(tf.ones([2, 2]))
+        self.assertFalse(
+            tf.reduce_all(tf.equal(new_model.weights[0], another_model.weights[0]))
+            .numpy()
+            .item()
+        )
+
+    def test_rel_path_from_pretrained(self):
+        model = DummyModel()
+        model(model.dummy_inputs)
+        model.save_pretrained(
+            f"tests/{WORKING_REPO_SUBDIR}/FROM_PRETRAINED",
+            config={"num": 10, "act": "gelu_fast"},
+        )
+
+        model = DummyModel.from_pretrained(
+            f"tests/{WORKING_REPO_SUBDIR}/FROM_PRETRAINED"
+        )
+        self.assertTrue(model.config == {"num": 10, "act": "gelu_fast"})
+
+    def test_abs_path_from_pretrained(self):
+        model = DummyModel()
+        model(model.dummy_inputs)
+        model.save_pretrained(
+            f"{WORKING_REPO_DIR}/{REPO_NAME}-FROM_PRETRAINED",
+            config={"num": 10, "act": "gelu_fast"},
+        )
+
+        model = DummyModel.from_pretrained(
+            f"{WORKING_REPO_DIR}/{REPO_NAME}-FROM_PRETRAINED"
+        )
+        self.assertDictEqual(model.config, {"num": 10, "act": "gelu_fast"})
+
+    def test_push_to_hub(self):
+        model = DummyModel()
+        model(model.dummy_inputs)
+        model.push_to_hub(
+            repo_path_or_name=f"{WORKING_REPO_DIR}/{REPO_NAME}-PUSH_TO_HUB",
+            api_endpoint=ENDPOINT_STAGING,
+            use_auth_token=self._token,
+            git_user="ci",
+            git_email="ci@dummy.com",
+            config={"num": 7, "act": "gelu_fast"},
+        )
+
+        model_info = self._api.model_info(
+            f"{USER}/{REPO_NAME}-PUSH_TO_HUB",
+        )
+        self.assertEqual(model_info.modelId, f"{USER}/{REPO_NAME}-PUSH_TO_HUB")
+
+        self._api.delete_repo(token=self._token, name=f"{REPO_NAME}-PUSH_TO_HUB")
+
+
+@require_tf
+class HubKerasSequentialTest(HubMixingTestKeras):
+    def model_init(self):
+        model = tf.keras.models.Sequential()
+        model.add(tf.keras.layers.Dense(2, activation="relu"))
+        model.compile(optimizer="adam", loss="mse")
+        return model
+
+    def test_save_pretrained(self):
+        model = self.model_init()
+
+        with pytest.raises(ValueError, match="Model should be built*"):
+            save_pretrained_keras(model, f"{WORKING_REPO_DIR}/{REPO_NAME}")
+
+        model.build((None, 2))
+
+        save_pretrained_keras(model, f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        files = os.listdir(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+
+        assert "saved_model.pb" in files
```

Comment:
nitpicking, but `assert` yields uninformative error messages when compared to the unittest's methods, such as `self.assertIn`, or `self.assertEqual`. Would it be possible to update those statements?